### PR TITLE
ref(superuser): only allow superuser write to modify admin options for project perf issue settings

### PR DESCRIPTION
**Pull request by cathteng**
Replace `is_active_superuser` in `ProjectPerformanceIssueSettings` with `superuser_has_permission`. Active superusers with the `auth:enterprise-superuser-read-write` feature flag must be a superuser with write permission in order to set modify admin options for project performance issue settings.

For https://github.com/getsentry/team-enterprise/issues/40

### CODE REVIEW

File: tests/sentry/api/endpoints/test_project_performance_issue_settings.py [hunk 1]
```diff
@@ -146,131 +134,166 @@ def test_get_project_options_overrides_threshold_defaults(self, get_value):
             assert response.data["consecutive_http_spans_min_time_saved_threshold"] == 1000
 
     def test_get_returns_error_without_feature_enabled(self):
-        with self.feature({}):
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_project_performance_issue_settings.py:149]: What was this even doing 🤔

File: tests/sentry/api/endpoints/test_project_performance_issue_settings.py [hunk 2]
```diff
@@ -146,131 +134,166 @@ def test_get_project_options_overrides_threshold_defaults(self, get_value):
...
+            status_code=status.HTTP_200_OK,
+        )
+
+        assert not response.data["n_plus_one_db_queries_detection_enabled"]
+
+        get_response = self.get_success_response(
+            self.project.organization.slug, self.project.slug, format="json"
+        )
 
         assert get_response.status_code == 200, response.content
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_project_performance_issue_settings.py:211]: Can move this into `get_success_response`

File: tests/sentry/api/endpoints/test_project_performance_issue_settings.py [hunk 3]
```diff
@@ -146,131 +134,166 @@ def test_get_project_options_overrides_threshold_defaults(self, get_value):
...
+            status_code=status.HTTP_200_OK,
+        )
 
-        assert response.status_code == 200, response.content
         assert response.data["n_plus_one_db_duration_threshold"] == 3000
 
-        with self.feature(PERFORMANCE_ISSUE_FEATURES):
-            get_response = self.client.get(self.url, format="json")
+        get_response = self.get_success_response(
+            self.project.organization.slug, self.project.slug, format="json"
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_project_performance_issue_settings.py:228]: nit: add 200 status code

File: tests/sentry/api/endpoints/test_project_performance_issue_settings.py [hunk 4]
```diff
@@ -146,131 +134,166 @@ def test_get_project_options_overrides_threshold_defaults(self, get_value):
...
+            status_code=status.HTTP_403_FORBIDDEN,
+        )
 
-        assert response.status_code == 403, response.content
         assert response.data == {"detail": "Disabled options can not be modified"}
 
-        with self.feature(PERFORMANCE_ISSUE_FEATURES):
-            get_response = self.client.get(self.url, format="json")
+        get_response = self.get_success_response(
+            self.project.organization.slug, self.project.slug, format="json"
```
Comment (schew2381) [inline tests/sentry/api/endpoints/test_project_performance_issue_settings.py:251]: nit: add 200 code

**Review by schew2381 — APPROVED** [review]
lgtm, small nits

File: tests/sentry/api/endpoints/test_project_performance_issue_settings.py (hunk 1 above)
Comment (cathteng) [reply tests/sentry/api/endpoints/test_project_performance_issue_settings.py:149]: idk

File: tests/sentry/api/endpoints/test_project_performance_issue_settings.py (hunk 3 above)
Comment (cathteng) [reply tests/sentry/api/endpoints/test_project_performance_issue_settings.py:228]: 200 is the default status?

Comment (schew2381) [reply tests/sentry/api/endpoints/test_project_performance_issue_settings.py:227]: Oh ok should be fine then

**Pull request merged**